### PR TITLE
Remove hack/e2e-internal.

### PR DESCRIPTION
**Pull request by spxtr**
Essentially, all that the scripts in `hack/e2e-internal` looked like this:

``` bash
KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..     

: ${KUBECTL:=${KUBE_ROOT}/cluster/kubectl.sh}       
: ${KUBE_CONFIG_FILE:="config-test.sh"}     

export KUBECTL KUBE_CONFIG_FILE     

source "${KUBE_ROOT}/cluster/kube-util.sh"      

prepare-e2e     

some-sort-of-command-from-kube-util-here
```

These scripts were only called from `hack/e2e.go`. I turned the environment-setting lines into `setEnvIfUnset`, and the final three lines into `utilCommand`.

### CODE REVIEW

**Comment by spxtr** [comment]
Looks like they only recently added `os.LookupEnv`, so I have to use the less-elegant `os.Getenv` for this if we want it to run on older go versions. Not the end of the world.

**Comment by spxtr** [comment]
@ixdy mind taking a look?

File: hack/e2e.go [hunk 1]
```diff
@@ -60,17 +60,13 @@ func absOrDie(path string) string {
...
-	Fail int
-}
-
-type ResultsByTest map[string]TestResult
-
 func main() {
 	log.SetFlags(log.LstdFlags | log.Lshortfile)
 	flag.Parse()
 
+	setEnvIfUnset("KUBECTL", *root+"/cluster/kubectl.sh"+kubectlArgs())
```
Comment (ixdy) [inline hack/e2e.go]: maybe use path.Join?

File: hack/e2e.go [hunk 2]
```diff
@@ -186,6 +175,22 @@ func Test() bool {
...
+		os.Setenv(env, value)
+	}
+}
+
+// Returns an exec.Cmd that sources cluster/kube-util.sh, runs prepare-e2e,
+// and then runs command.
+func utilCommand(command string) *exec.Cmd {
+	expandedCommand := os.ExpandEnv(command)
+	cmd := exec.Command("bash", "-s")
+	cmd.Stdin = strings.NewReader("source \"" + *root + "/cluster/kube-util.sh\" && prepare-e2e && " + expandedCommand)
```
Comment (ixdy) [inline hack/e2e.go]: again, maybe use path.Join?

File: hack/e2e.go [hunk 3]
```diff
@@ -83,17 +79,11 @@ func main() {
 	}
 
 	if *build {
-		// The build-release script needs stdin to ask the user whether
```
Comment (ixdy) [inline hack/e2e.go]: did we fix this? I can't remember anymore.

File: hack/e2e.go [hunk 4]
```diff
@@ -146,13 +135,13 @@ func Up() bool {
...
 	}
-	return finishRunning("up", exec.Command(path.Join(*root, "hack/e2e-internal/e2e-up.sh")))
+	return finishRunning("up", utilCommand("test-setup"))
 }
 
 // Ensure that the cluster is large engough to run the e2e tests.
 func ValidateClusterSize() {
 	// Check that there are at least minNodeCount nodes running
-	cmd := exec.Command(path.Join(*root, "hack/e2e-internal/e2e-cluster-size.sh"))
+	cmd := utilCommand("${KUBECTL} get nodes --no-headers | wc -l")
```
Comment (ixdy) [inline hack/e2e.go]: this whole function makes me cry

File: hack/e2e.go (hunk 3 above)
Comment (spxtr) [inline hack/e2e.go]: It used to be calling some subscript, but now it's not, so it already has stdin if it needs it.

**Comment by spxtr** [comment]
@ixdy PTAL

**Comment by spxtr** [comment]
Gonna give up on this now that there's a bunch of federated logic in there.

**Pull request closed without merging**